File: game/Unlimited.java
package game;

import game.utilities.HangmanEngine;
import game.utilities.SecureInput;

public class Unlimited {
    public static void runGame(){

        //Show Start Screen
        System.out.println("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");
        System.out.println("Unlimited Game:");

        //Initialize Engine
        HangmanEngine game = new HangmanEngine();

        //Start Game
        String guess = "";
        boolean gameOver = false;
        do {

            game.printGameState();

            System.out.print("Enter a letter: ");
            guess = SecureInput.secureStringChar();
            game.processGuess(guess);

            gameOver = game.isGuessed();

            System.out.println("\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n");

        } while(!gameOver);

        game.printPostGameState();

    }
}

File: game/unlimited.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include "utilities/hangman_engine.h"
#include "utilities/hangman_graphics_engine.h"
#include "utilities/secure_input.h"

static void clear_terminal(){
	printf("\033[2J\033[2J");
}

void run_unlimited_game(){
	int game_over = 0;
	char *guess;

	clear_terminal();
	printf("Unlimited Game:\n");
	hangman_engine_init();

	do{
		print_game_state();
		printf("Enter a letter: ");
		guess = secure_string();
		if(strlen(guess) == 1){
			free(guess);
		} else {
			free(guess);
		}

		game_over = !strcmp(hangman_engine_word, hangman_engine_guess_string);
		clear_terminal();
	} while(!game_over);

	print_post_game_state();
	printf("\nEnter any key to return to menu: ");
	fgetc(stdin);
	hangman_engine_exit();
}

